File: Goober.Web.Angular.Example/_src/grid-view-page/app/services/pagination-example/dtos/server-pagination-example-request.ts

import { FilterModelDto } from "@indusoft/grid-view";
import { SortModelDto } from "@indusoft/grid-view";


export class ServerPaginationExampleRequest {
    public pageNumber: number;
    public pageSize: number;
    public sortModel: SortModelDto;
    public filterModel: FilterModelDto;

    constructor(pageNumber: number, pageSize: number, filterModel: FilterModelDto, sortModel: SortModelDto) {
        this.pageNumber = pageNumber;
        this.pageSize = pageSize;
        this.filterModel = filterModel;
        this.sortModel = sortModel;
    }
}

File: Goober.Web.Angular.Example/Models/Api/ServerPaginationExample/ServerPaginationExampleRequest.cs
using Indusoft.Web.GridView.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Indusoft.Web.Angular.Example.Models.Api.ServerPaginationExample
{
    public class ServerPaginationExampleRequest
    {
        public int PageNumber { get; set; }
        public int PageSize { get; set; }
        public SortModelDto SortModel { get; set; }
        public FilterModelDto FilterModel { get; set; }
    }
}
